giles-eco/src/main/java/edu/asu/diging/gilesecosystem/web/core/service/delete/impl/DeleteDocumentService.java

```diff
@@ -0,0 +1,113 @@
+
+@Service
+public class DeleteDocumentService implements IDeleteDocumentService {
+    
+    public final static String REQUEST_PREFIX = "DELREQ";
+    
+    @Autowired
+    private IRequestFactory<IStorageDeletionRequest, StorageDeletionRequest> requestFactory;
+    
+    @Autowired
+    private ApplicationContext ctx;
+    
+    @Autowired
+    private ISystemMessageHandler messageHandler;
+    
+    @Autowired
+    private IRequestProducer requestProducer;
+    
+    @Autowired
+    private IPropertiesManager propertyManager;
+    
+    @Autowired
+    private ITransactionalFileService fileService;
+    
+    @Autowired
+    private ITransactionalUploadService uploadService;
+    
+    @Autowired
+    private ITransactionalDocumentService documentService;
+    
+    @Autowired
+    private ITransactionalProcessingRequestService processingRequestService;
+    
+    @PostConstruct
+    public void init() {
+        requestFactory.config(StorageDeletionRequest.class);
+    }
+    
+    @Override
+    @Async
+    public void initiateDeletion(IDocument document) throws GilesProcessingException, MessageCreationException {
+        IRequest storageDeletionRequest = createRequest(document);
+        requestProducer.sendRequest(storageDeletionRequest, propertyManager.getProperty(Properties.KAFKA_TOPIC_STORAGE_DELETION_REQUEST));
+    }
+    
+    @Override
+    public void completeDeletion(IDocument document) {
+        fileService.deleteFiles(document.getId());
+        processingRequestService.deleteRequestsByDocumentId(document.getId());
+        documentService.deleteDocument(document.getId());
+        deleteUpload(document.getUploadId());
+    }
+    
+    private void deleteUpload(String uploadId) {
+        // if an upload has multiple documents and only one of the documents is deleted the upload does not have to be deleted.
+        if(documentService.getDocumentsByUploadId(uploadId).isEmpty()) {
+            uploadService.deleteUpload(uploadId);
+        }
+    }
+    
+    private IRequest createRequest(IDocument document) throws GilesProcessingException {
+        IStorageDeletionRequest storageDeletionRequest = null;
+        try {
+            String requestId = documentService.generateRequestId(REQUEST_PREFIX);
+            storeRequestId(requestId, document);
+            storageDeletionRequest = requestFactory.createRequest(requestId, document.getUploadId());
+            storageDeletionRequest.setDocumentId(document.getId());
+        } catch (InstantiationException | IllegalAccessException e) {
+            throw new GilesProcessingException(e);
+        } 
+        return storageDeletionRequest;
+    }
+    
+    private void storeRequestId(String requestId, IDocument document) {
+        document.setRequestId(requestId);
+        try {
+            documentService.saveDocument(document);
+        } catch (UnstorableObjectException e) {
+            messageHandler.handleMessage("Could not store document.", e, MessageType.ERROR);
```

Comment:
if this is the case, should the rest of `createRequest` proceed? This looks like the exception should be thrown here.